src/concrete/ml/pytest/utils.py

```diff
@@ -388,6 +388,11 @@ def instantiate_model_generic(model_class, n_bits, **parameters):
         model_name (str): The type of the model as a string.
         model (object): The model instance.
     """
+    # Force multi threaded models to be single threaded as it is not working
+    # properly with pytest multi-threading
+    if "n_jobs" in model_class().get_params():
```

Comment:
I would prefer to make this less general like , isn't this parameter only for xgb ? if so I would just filter here "if xgb then n_jobs=1" as it's easier to maintain and debug in case we later have some issues related to this !

Reply:
No it's also for knn and random forest I believe. We really do not want any inteference with pytest multi threading. We don't really care about parallel training we only have very small dataset in our tests. It won't change anything.

Reply:
ok sure why not then !